### DISCUSSION TITLE
Incompatibility between react-day-picker and date-fns@4.1.0 causing installation failure

### DISCUSSION THREAD

**Discussion by amooonsen** [gpbl/react-day-picker#2632] [Support] [answered]
#### Description  
When attempting to install `react-day-picker` alongside `date-fns@4.1.0`, an error is thrown due to a dependency mismatch. The peer dependency required by `react-day-picker` specifies `date-fns` versions `^2.28.0 || ^3.0.0`, but the project requires `date-fns@4.1.0`.  

This dependency conflict makes it impossible to install `react-day-picker` without forcing the installation, potentially causing runtime issues.  

---

#### To Reproduce  

1. Install `react-day-picker` and `date-fns@4.1.0` in a React project:  
   ```bash
   npm install react-day-picker@8.10.1 date-fns@4.1.0
   ```

3. Use the following code in your project:  

   ```typescript
   import { DayPicker } from "react-day-picker";
   import "react-day-picker/style.css";

   export default function App() {
       return <DayPicker mode="single" />;
   }
   ```

4. Observe the error during dependency resolution or runtime.

Alternatively, use the provided CodeSandbox for easier reproduction:  
[[CodeSandbox link](https://codesandbox.io/p/sandbox/m2j7fr)](https://codesandbox.io/p/sandbox/m2j7fr)  

---

#### Actual Behavior  
The installation fails with the following error:  
```
npm error ERESOLVE unable to resolve dependency tree
npm error Could not resolve dependency:
npm error peer date-fns@"^2.28.0 || ^3.0.0" from react-day-picker@8.10.1
```

---

#### Expected Behavior  
The library should support newer versions of `date-fns` (e.g., `4.1.0`) or provide clear guidance on dependency compatibility.

---

#### Screenshots  
N/A  

---

#### Additional Information  

- **React version:** `18.2.0`  
- **react-day-picker version:** `8.10.1`  
- **date-fns version:** `4.1.0`  
- **Node version:** `18.x`  

---

#### Suggested Solutions  

1. Update `react-day-picker` to support `date-fns@4.x`.  
2. Provide alternative recommendations if `date-fns@4.x` compatibility is not intended.  

Let me know if additional details are required!

**Comment by gpbl** [comment, marked as answer]
Hi @amooonsen! Please upgrade to v9 for `date-fns@4` support, or use `overrides` in your package.json https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides (I can't guarantee it works!)
